lnbits/extensions/boltz/boltz.py

```diff
@@ -0,0 +1,458 @@
+    # hardcoded maximum tx size, in the future we try to get the size of the tx via embit (not possible yet)
+    tx_size_vbyte = 200
+    mempool_fees = get_mempool_fees()
+    return mempool_fees * tx_size_vbyte
+
+
+# claim tx for reverse swaps
+# refund tx for normal swaps
+async def create_onchain_tx(
+    swap: Union[ReverseSubmarineSwap, SubmarineSwap], mempool_lockup_tx
+) -> Transaction:
+
+    if type(swap) == SubmarineSwap:
+        current_block_height = get_mempool_blockheight()
+        if current_block_height <= swap.timeout_block_height:
+            msg = f"refund not possible, timeout_block_height ({swap.timeout_block_height}) is not yet exceeded ({current_block_height})"
+            raise HTTPException(status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail=msg)
+        privkey = ec.PrivateKey.from_wif(swap.refund_privkey)
+        preimage = b""
+        onchain_address = swap.refund_address
+        sequence = 0xFFFFFFFE
+    else:
+        privkey = ec.PrivateKey.from_wif(swap.claim_privkey)
+        preimage = unhexlify(swap.preimage)
+        onchain_address = swap.onchain_address
+        sequence = 0xFFFFFFFF
+
+    locktime = swap.timeout_block_height
+    redeem_script = unhexlify(swap.redeem_script)
+
+    fees = get_fee_estimation()
+
+    tx, txid, vout_cnt, vout_amount = mempool_lockup_tx
+
+    script_pubkey = script.address_to_scriptpubkey(onchain_address)
+
+    vin = [TransactionInput(unhexlify(txid), vout_cnt, sequence=sequence)]
+    vout = [TransactionOutput(vout_amount - fees, script_pubkey)]
+    tx = Transaction(vin=vin, vout=vout)
+
+    if type(swap) == SubmarineSwap:
+        tx.locktime = locktime
+
+    # TODO: 2 rounds for fee calculation, look at vbytes after signing and do another TX
+    s = script.Script(data=redeem_script)
+    for i in range(len(vin)):
+        inp = vin[i]
+        if type(swap) == SubmarineSwap:
+            # OP_0 == 0
+            # OP_PUSHDATA34 == 34
+            # OP_PUSHDATA35 == 35
+            rs = bytes([34]) + bytes([0]) + bytes([32]) + sha256(redeem_script).digest()
+            tx.vin[i].script_sig = script.Script(data=rs)
+        h = tx.sighash_segwit(i, s, vout_amount)
+        sig = privkey.sign(h).serialize() + bytes([SIGHASH.ALL])
+        witness_items = [sig, preimage, redeem_script]
+        tx.vin[i].witness = script.Witness(items=witness_items)
+
+    return tx
+
+
+def getTimestamp():
+    date = datetime.datetime.utcnow()
+    return calendar.timegm(date.utctimetuple())
+
+
+def create_post_request(url, payload={}):
+    payload["referralId"] = "lnbits"
+    res = httpx.post(
+        url,
+        json=payload,
+        headers={"Content-Type": "application/json"},
+        timeout=40,
+    )
+    handle_request_errors(res)
+    return res.json()
+
+
+def create_get_request(url):
+    res = httpx.get(
+        url,
+        headers={"Content-Type": "application/json"},
+        timeout=40,
+    )
+    handle_request_errors(res)
+    return res.json()
```

Comment:
This function is only used once, maybe un-factor into `get_boltz_pairs`.

Reply:
dd451e1d